### DISCUSSION TITLE
`dnf builddep` as info only, not to install packages

### DISCUSSION THREAD

**Discussion by brianjmurrell** [rpm-software-management/dnf5#1662] [Q&A]
I wonder if there is any way to effect simply outputting an RPM specfiles's `BuildRequires:`, similar to how `dnf builddep` works but without actually trying to install the packages, but rather just report on what they are?

**Comment by jan-kolarik** [comment]
Hi Brian, 

If it's an RPM in the repository, you can query its build requirements with a command like `dnf5 repoquery --disablerepo="*" --enablerepo="*-source" --arch=src --requires <package_name>`. 

If you only have the spec file, I think there currently isn't a way to get machine-readable output for that. You can try `dnf5 builddep <spec> --assumeno`, which will show the builddep output without actually installing anything, but that might not be exactly what you're looking for.

There will be improvements to the output soon with https://github.com/rpm-software-management/dnf5/pull/1641, which should simplify it to just the transaction table on the stdout. If you would like a JSON output for the transaction table, we could plan to implement it with https://github.com/rpm-software-management/dnf5/issues/867 in the future.

**Comment by m-blaha** [comment]
Allow me to provide more details on the usage of `dnf5 builddep --assumeno`: only the packages that are missing on the system where the command is executed are included in the transaction table. The build requirements that are already installed are printed separately (with a message like `Package XXX is already installed`), and I wouldn’t expect them to appear in the transaction table, even in JSON format. Therefore, it's not a universal tool for retrieving a list of build requirements for a given spec. The results are always specific to the current system state.

**Reply by jan-kolarik** [reply]
I guess you could workaround this by pointing the dnf to an empty installroot, something like adding `--installroot <path_to_empty_dir> --use-host-config` ?

**Reply by m-blaha** [reply]
That might work. Also be aware, that also all the build requirements dependencies will be part of the transaction table (in the `Installing (weak) dependencies` section).

**Comment by ppisar** [comment]
You don't need dnf for that. It's an RPM thing, so use rpm:

* `rpm -q --requires -p SOURCE_PACKAGE.rpm`
* `rpmspec -q --buildrequires PACKAGE.spec`

**Reply by m-blaha** [reply]
Indeed. But dnf tool might have an advantage of resolving requirement strings to specific packages. But it depends on what is the use case.

@brianjmurrell could you elaborate more on the use case you are dealing with?
